src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.Data.cs

```diff
@@ -17,11 +17,98 @@ namespace Microsoft.NetCore.Analyzers.InteropServices
 {
     using ValueContentAnalysisResult = DataFlowAnalysisResult<ValueContentBlockAnalysisResult, ValueContentAbstractValue>;
 
-    public sealed partial class RuntimePlatformCheckAnalyzer
+    public sealed partial class PlatformCompatabilityAnalyzer
     {
-        private struct RuntimeOSPlatformInfo : IAbstractAnalysisValue, IEquatable<RuntimeOSPlatformInfo>
+        private enum PlatformAttributeType
         {
-            private RuntimeOSPlatformInfo(string invokedPlatformCheckMethodName, string platformPropertyName, Version version, bool negated)
+            None, MinimumOSPlatformAttribute, ObsoletedInOSPlatformAttribute, RemovedInOSPlatformAttribute, TargetPlatformAttribute
+        }
+
+        private struct PlatformAttributeInfo : IEquatable<PlatformAttributeInfo>
+        {
+            public PlatformAttributeType AttributeType { get; set; }
+            public string PlatformName { get; set; }
+            public Version Version { get; set; }
+
+            public static bool TryParsePlatformAttributeInfo(AttributeData osAttribute, out PlatformAttributeInfo parsedAttribute)
+            {
+                parsedAttribute = new PlatformAttributeInfo();
+                switch (osAttribute.AttributeClass.Name)
+                {
+                    case MinimumOSPlatformAttribute:
```

Comment:
We should not be comparing attribute names. Instead, we should fetch the well-known type symbol for each attribute upfront and perform symbol equality check.

Reply:
We will be adding the platform attributes into ref assemblies of older versions of .NetCore and netstandard etc., where the attributes will be internal, so while diagnosing APIs used from those libraries we will not have access to the attributes, so we need to / can check only names

Reply:
@buyaa-n I am not suggesting depending on the attribute or the type. I am suggesting we add a new entry in `WellKnownTypeNames` for each attribute and then fetch the type using `compilation.GetOrCreateTypeByMetadataName` upfront. We do the same for all analyzers ensuring there is no actual dependency on the analyzer type.

Reply:
Sorry, my explanation was not completely correct, because we are adding the new attributes into older versions of .Net 5.0,  there could be multiple referenced assemblies each with their own version of this internal attribute type. Talked with @mavasani offline, we gonna add namespace check in addition to type name check. (by new design come out early this week the namespaces also gonna change, so I will not add that until they changed)

Reply:
Thanks for the clarification @buyaa-n! Please add a descriptive comment in the core analyzer file above the block of const attribute name definitions for reference. If possible, also add a test that runs for this scenario, i.e. attribute has multiple identical definitions in the compilation and it still passes.

Reply:
Adding a comment, for testing not sure how to test with unit test. I run the analyzer in runtime, which could be real-world testing for this scenario

Reply:
> not sure how to test with unit test

You can write a test that is running against .net that defines these public attributes + also use your mock attribute in source. So basically, you will have 2 copies of these attributes. The test will only pass if you name comparison based approach for attributes.

Reply:
> You can write a test that is running against .net that defines these public attributes + also use your mock attribute in test input source. So basically, you will have 2 copies of these attributes. The test will only pass if you use name comparison based approach for attributes.

Test are already targeting 5.0 plus also using mock attributes, so i would assume this is covered, https://github.com/dotnet/roslyn-analyzers/blob/3916447c7269cdb91a7de6e84355592e04451986/src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzerTests.cs#L1088